core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/ComponentParser.java

```diff
@@ -0,0 +1,269 @@
+            INSTANCE = new ComponentParser();
+        }
+        return INSTANCE;
+    }
+
+    public List<Component> parse(OpenAPI openAPI) {
+        List<Component> components = new ArrayList<>();
+
+        securitySchemeMap = openAPI.getComponents().getSecuritySchemes();
+        versionNumber = Long.parseLong(openAPI.getInfo().getVersion());
+        connectionName = openAPI.getInfo().getTitle();
+
+        Paths paths = openAPI.getPaths();
+        for (String pathName : paths.keySet()) {
+            PathItem path = paths.get(pathName);
+            Map<PathItem.HttpMethod, Operation> operations = path.readOperationsMap();
+
+            for (Map.Entry<PathItem.HttpMethod, Operation> entry : operations.entrySet()) {
+                components.addAll(initComponents(entry.getValue(), entry.getKey(), pathName));
+            }
+
+        }
+
+       return components;
+    }
+
+    public List<Component> initComponents(Operation operation, PathItem.HttpMethod operationName, String pathName) {
+        List<Component> components = new ArrayList<>();
+        List<String> securities = new ArrayList<>(getSecurities(operation));
+        List<String> requestContents = new ArrayList<>(getRequestContents(operation.getRequestBody()));
+        List<String> responseContents = new ArrayList<>(getResponseContents(operation.getResponses()));
+        List<List<String>> nameCombinations = new ArrayList<>(Arrays.asList(securities, requestContents, responseContents));
+        List<HashMap<String, String>> names = new ArrayList<>();
+
+        if (!(securities.isEmpty() && requestContents.isEmpty() && responseContents.isEmpty())) {
+            names = generateNames(nameCombinations, new ArrayList<>(), 0, new LinkedHashMap<>());
+        }
+
+
+
+        if (!names.isEmpty()) {
+            for (HashMap<String, String> partNames : names) {
+                components.add(createComponent(partNames, operation, operationName, pathName));
+            }
+        } else {
+            components.add(createComponent(new HashMap<>(), operation, operationName, pathName));
+        }
+        return components;
+    }
+
+
+    public Set<String> getSecurities(Operation operation) {
+        Set<String> securities = new HashSet<>();
+        List<Parameter> parameters = operation.getParameters();
+        List<SecurityRequirement> securityRequirements = operation.getSecurity();
+
+        if (parameters != null) {
+            securities = parameters.stream().map(parameter -> {
+                if (securitySchemeMap.containsKey(parameter.getName())) {
+                    return parameter.getName();
+                }
+                return null;
+            }).filter(Objects::nonNull).collect(Collectors.toSet());
+        }
+
+        if (securityRequirements != null) {
+            securities.addAll(securityRequirements.stream().map(securityRequirement -> (String) securityRequirement.keySet().toArray()[0]).collect(Collectors.toSet()));
```

Comment:
what does '(String) securityRequirement.keySet().toArray()[0])' return exactly?

Reply:
If we have these securities :       
security:
        - petstore_auth:
            - write:pets
            - read:pets
        - user_auth:
          - write:pets
 securityRequirements has two securityRequirement [ SecurityRequirement, SecurityRequirement ]
Regarding the library's implementation, the SecurityRequirement is an HashMap. I tested several scenarios and it has only one key that is the name of the security.

To get the name I just got the keyset array that should normally have one key. I took this key at index 0.